bin/cliDownload.js

```diff
@@ -0,0 +1,75 @@
+import fs from "fs";
+import path from "path";
+import { DicomAccess } from "../src/access/DicomAccess.js";
+import { DicomStoreSDW } from "../src/store/DicomStoreSDW.js";
+
+const action = async (url, options) => {
+  // Create download directory using StudyInstanceUID
+  const baseDir = path.join(options.directory, options.StudyInstanceUID);
+  let downloadDir = baseDir;
+  let counter = 1;
+
+  // Ensure directory does not already exist; add suffix if needed
+  while (fs.existsSync(downloadDir)) {
+    downloadDir = `${baseDir}_${counter++}`;
+  }
+
+  fs.mkdirSync(downloadDir, { recursive: true });
+  console.log(`📁 Study will be downloaded to: ${downloadDir}`);
+
+  try {
+    // Create access instance (currently supports only DICOMweb)
+    console.log("🔌 Creating DICOM access instance...");
```

Comment:
The download action should basically look like:

const access = await DicomAccess.createInstance(url, options);
const destination = await DicomAccess.createInstance(baseDir, options)

const studyAccess = await access.queryStudy()
const studyDest = await access.queryStudy({ create: true })

studyDest.writeStudy(studyAccess);

Then, internally studyAccess should have things iterated from/over that studyDest needs in order to write the study source.

That way, studyDest being a DICOMDIR containing part 10 files can be written to, as can a response to a DICOMweb part 10 query, or a DICOMweb query can all be written to independently of the source type.